lib/command.js

```diff
@@ -76,21 +76,45 @@ const parseExpression = expression => {
 	throw new TypeError(`Unexpected "${typeOfExpression}" in template expression`);
 };
 
-const parseTemplate = (template, index, templates, expressions) => {
+const concatTokens = (tokens, nextTokens, isNew) => isNew || tokens.length === 0 || nextTokens.length === 0
+	? [...tokens, ...nextTokens]
+	: [
+		...tokens.slice(0, -1),
+		`${tokens[tokens.length - 1]}${nextTokens[0]}`,
+		...nextTokens.slice(1),
+	];
+
+const parseTemplate = ({templates, expressions, tokens, index, template}) => {
 	const templateString = template ?? templates.raw[index];
 	const templateTokens = templateString.split(SPACES_REGEXP).filter(Boolean);
+	const newTokens = concatTokens(
+		tokens,
+		templateTokens,
+		templateString.startsWith(' '),
+	);
 
 	if (index === expressions.length) {
-		return templateTokens;
+		return newTokens;
 	}
 
 	const expression = expressions[index];
+	const expressionTokens = Array.isArray(expression)
+		? expression.map(expression => parseExpression(expression))
+		: [parseExpression(expression)];
+	return concatTokens(
+		newTokens,
+		expressionTokens,
+		templateString.endsWith(' '),
+	);
+};
+
+export const parseTemplates = (templates, expressions) => {
```

Comment:
We could use `Array.reduce()` too, but it is forbidden by xo, so I opted for a `for` loop instead.